src/data_types/strs.rs

```diff
@@ -161,6 +163,44 @@ impl CStr16 {
             pos: 0,
         }
     }
+
+    /// Write a string slice into the provided buffer. If it fails, then most probably, because
+    /// the buffer is not big enough. In that case, the buffer will contain the correct string
+    /// until the point, where the size was not enough.
+    ///
+    /// ## Example
+    /// ```rust
```

Comment:
Minor nitpick: let's leave a blank line `///` between the title of the section and its contents.